components/support/nimbus-fml/src/workflows.rs

```diff
@@ -158,6 +162,109 @@ pub(crate) fn fetch_file(files: &LoaderConfig, nm: &str) -> Result<()> {
     Ok(())
 }
 
+fn output_ok(term: &Term, title: &str) -> Result<()> {
+    let style = term.style().green();
+    term.write_line(&format!("✅ {}", style.apply_to(title)))?;
+    Ok(())
+}
+
+fn output_note(term: &Term, title: &str) -> Result<()> {
+    let style = term.style().yellow();
+    term.write_line(&format!("ℹ️ {}", style.apply_to(title)))?;
+    Ok(())
+}
+
+fn output_err(term: &Term, title: &str, detail: &str) -> Result<()> {
+    let style = term.style().red();
+    term.write_line(&format!("❎ {}: {detail}", style.apply_to(title),))?;
+    Ok(())
+}
+
+pub(crate) fn validate(cmd: &ValidateCmd) -> Result<()> {
+    let term = Term::stdout();
+
+    let files: FileLoader = TryFrom::try_from(&cmd.loader)?;
+
+    let filename = &cmd.manifest;
+    let file_path = files.file_path(filename)?;
+    let parser: Parser = Parser::new(files, file_path.clone())?;
+    let mut loading = HashSet::new();
+    let manifest_front_end = parser.load_manifest(&file_path, &mut loading)?;
+
+    let iter_includes = loading.iter().map(|id| id.to_string());
+
+    let channels = manifest_front_end.channels();
+    if channels.is_empty() {
+        output_note(
+            &term,
+            &format!(
+                "Loaded modules:\n- {}\n",
+                iter_includes.collect::<Vec<String>>().join("\n- ")
+            ),
+        )?;
+        output_ok(&term, &format!(
+            "{}\n{}\n{}",
+            "The manifest is valid for including in other files. To be imported, or used as an app manifest, it requires the following:",
+            "- A `channels` list",
+            "- An `about` block",
+        ))?;
+        return Ok(());
+    }
+    let intermediate_representation = parser.get_intermediate_representation(&channels[0])?;
```

Comment:
It would be nice if we can get this to fail with some indication of which file is being processed.

Or at least, output the error with an `output_err`.

Reply:
Yeah I think I'm going to add an optional manifests hashmap param to that method so that we _can_ get the imports in order and keep a record of which ones were imported and where it failed and stuff.

Reply:
And maybe rework the iteration so it keeps iterating but records successes/failures

Reply:
But I'm going to handle that in a separate ticket (after updating that error to go to `output_err`)

Reply:
https://mozilla-hub.atlassian.net/browse/EXP-3551